persist/fs/commitlog/reader.go

```diff
@@ -182,36 +339,44 @@ func (r *reader) readChunk() ([]byte, error) {
 	if _, err := r.chunkReader.Read(buffer); err != nil {
 		return nil, err
 	}
-	return buffer, nil
+
+	// Copy message into pooled byte slice. Reading happens synchronously so its
+	// safe to reuse the same buffer repeatedly, but decoding happens in parallel
+	// so we need to make sure that the byte slice we return won't be mutated
+	// until the caller is done with it.
+	pooled := r.bytesPool.Get(int(size))
+	pooled.IncRef()
+	pooled.AppendAll(buffer)
+	return pooled, nil
 }
 
 func (r *reader) readInfo() (schema.LogInfo, error) {
 	data, err := r.readChunk()
 	if err != nil {
 		return emptyLogInfo, err
 	}
-	r.logDecoder.Reset(data)
-	return r.logDecoder.DecodeLogInfo()
+	r.logDecoder.Reset(data.Get())
+	logInfo, err := r.logDecoder.DecodeLogInfo()
+	data.DecRef()
```

Comment:
This seems to leak the `checked.Bytes` here, need to `data.Finalize()` I believe.